File: android/app/src/main/java/com/example/aranimation/model/ARModelItem.kt
package com.example.aranimation.model

/**
 * Đại diện cho một phần tử mô hình 3D trong danh sách lựa chọn AR.
 *
 * @property id Định danh duy nhất cho mô hình
 * @property displayName Tên hiển thị thân thiện với người dùng
 * @property assetPath Đường dẫn tệp .glb bên trong thư mục `src/main/assets/`
 * @property isSelected Trạng thái đang được chọn trong danh sách hay không
 */
data class ARModelItem(
    val id: String,
    val displayName: String,
    val assetPath: String,
    var isSelected: Boolean = false
) {
    companion object {
        /**
         * Danh sách 6 mô hình động vật 3D mặc định từ assets.
         * Mặc định chọn con vật đầu tiên ("stag.glb").
         */
        fun getDefaultList(): List<ARModelItem> = listOf(
            ARModelItem(
                id = "stag",
                displayName = "Stag",
                assetPath = "stag.glb",
                isSelected = true
            ),
            ARModelItem(
                id = "wolf",
                displayName = "Wolf",
                assetPath = "wolf.glb",
                isSelected = false
            ),
            ARModelItem(
                id = "bull",
                displayName = "Bull",
                assetPath = "bull.glb",
                isSelected = false
            ),
            ARModelItem(
                id = "cow",
                displayName = "Cow",
                assetPath = "cow.glb",
                isSelected = false
            ),
            ARModelItem(
                id = "deer",
                displayName = "Deer",
                assetPath = "deer.glb",
                isSelected = false
            ),
            ARModelItem(
                id = "shiba",
                displayName = "Shiba",
                assetPath = "shiba.glb",
                isSelected = false
            )
        )
    }
}

File: ios/ARAnimationApp/Models/ARModelItem.swift
import Foundation

/// Đại diện cho một mô hình 3D AR trên hệ điều hành iOS (RealityKit / QuickLook).
///
/// Tuân thủ `Identifiable` để dễ dàng duyệt trong SwiftUI `ForEach` hoặc `List`,
/// và `Equatable` để kiểm tra trạng thái item đang được chọn.
struct ARModelItem: Identifiable, Equatable {
    /// Định danh duy nhất của mô hình
    let id: String
    
    /// Tên hiển thị thân thiện trên giao diện người dùng
    let displayName: String
    
    /// Tên tệp .usdz được đóng gói trong Main Bundle của ứng dụng
    let fileName: String
    
    /// Khởi tạo một phần tử mô hình AR
    init(id: String, displayName: String, fileName: String) {
        self.id = id
        self.displayName = displayName
        self.fileName = fileName
    }
}

extension ARModelItem {
    /// Danh sách mẫu 6 mô hình động vật 3D định dạng .usdz
    static let sampleAnimals: [ARModelItem] = [
        ARModelItem(id: "stag", displayName: "Stag", fileName: "stag.usdz"),
        ARModelItem(id: "wolf", displayName: "Wolf", fileName: "wolf.usdz"),
        ARModelItem(id: "bull", displayName: "Bull", fileName: "bull.usdz"),
        ARModelItem(id: "cow", displayName: "Cow", fileName: "cow.usdz"),
        ARModelItem(id: "deer", displayName: "Deer", fileName: "deer.usdz"),
        ARModelItem(id: "shiba", displayName: "Shiba", fileName: "shiba.usdz")
    ]
    
    /// Mô hình con vật được chọn mặc định ban đầu ("stag.usdz")
    static var defaultAnimal: ARModelItem {
        sampleAnimals.first ?? ARModelItem(id: "stag", displayName: "Stag", fileName: "stag.usdz")
    }
}
